pkg/restclient/peer.go

```diff
@@ -0,0 +1,27 @@
+package restclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gluster/glusterd2/pkg/api"
+)
+
+// PeerProbe adds a peer to the Cluster
+func (c *RESTClient) PeerProbe(host string) error {
+	peerAddReq := api.PeerAddReq{
+		Addresses: []string{host},
+	}
+	reqBody, err := json.Marshal(peerAddReq)
+	if err != nil {
+		return err
+	}
+	return httpRESTAction("POST", c.baseURL+"/v1/peers", strings.NewReader(string(reqBody)), 201)
```

Comment:
General comment for this and the rest of the client methods that have been implemented.

You could do much more stuff inside `httpRestAction`.
- If you pass in the `RestClient` or make `httpRestAction` a method on `RestClient` you can form the URL inside, instead of concatenating here.
- You could just pass the request struct in here, and do the marshalling in `httpRESTAction`.
- You can avoid passing the expected code, you can have a map of action and expected code and use it instead.

Also, use the `http.Status*` enums for status codes, instead of the values directly.